docs/nat.md

```diff
@@ -0,0 +1,61 @@
+# NAT Extension for ENRs
+
+This document details the extra fields added to ENRs to minimally support NAT
+hole-punching in [discv5](https://github.com/sigp/discv5).
+
+
+## Extended Fields
+
+This library supports the following additional fields:
+
+| Key | Value |
+| --- | ----- |
+| `features` | A bitfield representing which features are supported |
+| `nat` | IPv4 address, 4 bytes, representing a NAT'd external IP address |
+| `nat6` | IPv6 address, 16 bytes, representing a NAT'd external IPv6 address |
+
+### Features
+
+This field is a generic bitfield that allows ENRs to indicate which features
+they support. The current supported features are:
+- 1 - NAT_SUPPORT: Setting the first bit (big-endian) indicates that this node
+	supports NAT hole-punching. This means it can act as a relayer to
+	facilitate hole punching between other nodes that also support this
+	feature.
+
+Features may be added in the future.
+
+### NAT
+
+The `nat` and `nat6` fields are used to indicate how an external peer should
+attempt a connection to the local node. Setting these fields, indicate that the
+local node has identified itself as being behind a NAT and a hole-punching
+technique should be used to contact it.
```

Comment:
Regarding my previous comment, I think it would be good to specify that declaring a nat related field also means supporting nat hole punching to itself, even if the NAT_FEATURE is off (so, even if not being a relayer to others). There is otherwise no point in advertising the field